### ISSUE TITLE
Reg: Issue in using Parchment.

### ISSUE THREAD

**Issue by prasanna1211** [quilljs/quill#1024]
Hy,
I am pretty much new with quill. I was trying to use blots for one of my purpose. It solved all my problems except one feature. Basically I want the essential functionalities of editor along with one feature. The feature is there will be a separate popup which has buttons. Each button will have a text. On clicking this button, the editor will print the text along with an close option. Something similar to facebook mentions. Also when I click the close button that entire text along with the close button should get erased. This is the code which I implemented.

Just added the JS code in the link

http://codepen.io/anon/pen/pEdwrx

insertVariable will be called whenever a button on the popup is clicked and for that particular text I am adding something like <span class="proc-link" style="text-decoration: underline" variable="value1">First Name<span class="close">X</span></span>.

Basically I wrap the text in a span and I wrap a X inside the span with a class and I nest this span inside the parent span. So that I can use JQUERY to remove the parent of X when X is clicked.

The problem which I get is when I insert the text and after that I type something it is typing along with X. It doesn't create a new <p> after the span. Basically I am not able to set the selection outside of the span. Is there a way to do it ?

**Comment by prasanna1211** [comment]
I think forgetting the code it is a functionality. When a button is clicked it remains active unless we again click it. Which function is used to change from active state to inactive state ?

**Comment by jhchen** [comment]
The CodePen example does not run.
